sdk/src/main/scala/io/horizen/account/state/MessageProcessorUtil.scala

```diff
@@ -2,16 +2,20 @@ package io.horizen.account.state
 
 import io.horizen.account.sc2sc.ScTxCommitmentTreeRootHashMessageProcessor
 import io.horizen.cryptolibprovider.CircuitTypes.{NaiveThresholdSignatureCircuit, NaiveThresholdSignatureCircuitWithKeyRotation}
+import io.horizen.fork.{ForkManager, Sc2ScFork}
 import io.horizen.params.NetworkParams
 import io.horizen.sc2sc.Sc2ScUtils
+import io.horizen.utils.TimeToEpochUtils
+import sparkz.core.utils.TimeProvider
 
 object MessageProcessorUtil {
-  def getMessageProcessorSeq(params: NetworkParams, customMessageProcessors: Seq[MessageProcessor]): Seq[MessageProcessor] = {
+  def getMessageProcessorSeq(params: NetworkParams, customMessageProcessors: Seq[MessageProcessor], timeProvider: TimeProvider): Seq[MessageProcessor] = {
     val maybeKeyRotationMsgProcessor = params.circuitType match {
       case NaiveThresholdSignatureCircuit => None
       case NaiveThresholdSignatureCircuitWithKeyRotation => Some(CertificateKeyRotationMsgProcessor(params))
     }
-    val sc2ScMsgProcessors = if (Sc2ScUtils.isActive(params)) Seq(ScTxCommitmentTreeRootHashMessageProcessor())
+    val epoch = TimeToEpochUtils.timeStampToEpochNumber(params, timeProvider.time())
+    val sc2ScMsgProcessors = if (Sc2ScUtils.isActive(params, ForkManager.getOptionalSidechainFork[Sc2ScFork](epoch))) Seq(ScTxCommitmentTreeRootHashMessageProcessor())
```

Comment:
As an alternative to this approach, instead of adding the message processor to the list depending on fork point, the list could be made static, and the fork handling could be left to the init() / canProcess() methods of the specific (fork dependant) message processor.